LocalMultiplayerAgent/DeploymentTool/DeploymentScript.cs

```diff
@@ -0,0 +1,270 @@
+                    CreateBuildWithManagedContainerRequest request = GetManagedContainerRequest();
+
+                    foreach (var file in request.GameAssetReferences)
+                    {
+                        await CheckAssetFiles(file.FileName);
+                    }
+
+                    createBuild = await CreateBuildWithManagedContainer(request);
+                }
+                else
+                {
+                    //TODO: for CustomLinux Container
+                }
+            }
+            else
+            {
+                CreateBuildWithProcessBasedServerRequest request = GetProcessBasedServerRequest();
+
+                foreach (var file in request.GameAssetReferences)
+                {
+                    await CheckAssetFiles(file.FileName);
+                }
+
+                createBuild = await CreateBuildWithProcessBasedServer(request);
+            }
+
+            if (createBuild.Error != null)
+            {
+                foreach (var error in createBuild.Error.ErrorDetails)
+                {
+                    foreach (var errorMessage in error.Value)
+                    {
+                        Console.WriteLine($"{errorMessage}");
+                    }
+                }
+            }
+        }
+
+        public CreateBuildWithCustomContainerRequest GetCustomContainerRequest()
+        {
+            return new CreateBuildWithCustomContainerRequest
+            {
+                BuildName = deploymentSettings.BuildName,
+                VmSize = Enum.Parse<AzureVmSize>(deploymentSettings.VmSize),
+                ContainerFlavor = ContainerFlavor.CustomLinux,
+                ContainerImageReference = new ContainerImageReference()
+                {
+                    ImageName = settings.ContainerStartParameters.ImageDetails.ImageName,
+                    Tag = settings.ContainerStartParameters.ImageDetails.ImageTag
+                },
+                Ports = PortMapping(),
+                ContainerRunCommand = settings.ContainerStartParameters.StartGameCommand,
+                RegionConfigurations = deploymentSettings.RegionConfigurations?.Select(x => new BuildRegionParams()
+                {
+                    Region = x.Region,
+                    MaxServers = x.MaxServers,
+                    StandbyServers = x.StandbyServers,
+                    MultiplayerServerCountPerVm = deploymentSettings.MultiplayerServerCountPerVm,
+                    VmSize = Enum.Parse<AzureVmSize>(deploymentSettings.VmSize)
+
+                }).ToList(),
+                MultiplayerServerCountPerVm = deploymentSettings.MultiplayerServerCountPerVm,
+            };
+        }
+
+        public CreateBuildWithManagedContainerRequest GetManagedContainerRequest()
+        {
+            return new CreateBuildWithManagedContainerRequest
+            {
+                VmSize = Enum.Parse<AzureVmSize>(deploymentSettings.VmSize),
+                GameCertificateReferences = null,
```

Comment:
Is there a plan to enable developers to use this tool to upload certificates as well? Or if a developer is using certificates will they have to use the portal?